ruby/lint/action.yml

```diff
@@ -22,3 +22,6 @@ runs:
         name: "${{ github.sha }}-lint-results.out"
         path: ./rubocop/rubocop.json
         retention-days: 1
+    - name:
+      shell: bash
+      run: exit ${{ steps.rubocop.outputs.exit-code }}
```

Comment:
Still not fully clear on why we need this. We can let this step exit whatever the rubocop exit code is straight up, as long as the step is the last step in a job and not part of another job.

Reply:
It preserves consistency with what the team had set for code coverage and test result [artifact uploading](https://github.com/wishabi/github-actions/blob/main/go/test/action.yml#L19) in Go. We've been uploading artifacts in our actions after running the command and here the rubocop output file is being uploaded as an artifact for use with SonarQube later. If we don't continue on error, upload the artifact and then exit on this line, the SonarQube scan will fail with a file not found. 

The difference with the tests is we are failing the workflow if our tests fail (no need to keep running), but we're currently ignoring the lint results in `search-sieve`. We should definitely strive to get our lint rules and code to the point that we'd fail if the linter fails.

If we're concerned about coupling, which is a potential issue as well, then there's a few options (probably easier to chat).

Reply:
Hmm... I think the Sonarqube step should depend on the lint step. If the lint step fails, the Sonarqube one wouldn't even run, so it wouldn't fail :smile: Unless I'm missing something.

Reply:
I'm happy to fail it completely if linting fails with our rubocop config - I was just copying the functionality that exists on `search-sieve` currently. I wasn't sure the status of our lint profile and unless we add a little scope creep starting to fail a build on lint failure **will break** the `search-sieve` build.